conf/obj.h

```diff
@@ -716,29 +717,43 @@ struct m0_conf_drive {
 
 /** Filters group. All filters in group match record type they work with. */
 struct m0_conf_fdmi_flt_grp {
-	struct m0_conf_obj  ffg_obj;
+	struct m0_conf_obj        ffg_obj;
 	/* Type of the record that filters of this group support. */
-	uint32_t            ffg_rec_type;
+	enum m0_fdmi_rec_type_id  ffg_rec_type;
 	/* Directory of filters for this filters group. */
-	struct m0_conf_dir *ffg_filters;
+	struct m0_conf_dir       *ffg_filters;
 };
 
 struct m0_conf_fdmi_filter {
-	struct m0_conf_obj     ff_obj;
-	struct m0_fid          ff_filter_id;
-	struct m0_fdmi_filter  ff_filter;
+	struct m0_conf_obj            ff_obj;
+	enum m0_fdmi_filter_type_id   ff_type;
+	struct m0_fid                 ff_filter_id;
+	struct m0_fdmi_filter         ff_filter;
 	/** The node this filter is hosted at. */
-	struct m0_conf_node   *ff_node;
+	struct m0_conf_node          *ff_node;
+	/**
+	 * DIX pool version to search kv pairs for substrings for.
+	 * The filter would match a CAS operation iff the pool version for the
+	 * operation matches.
+	 * It's only used when ff_type == M0_FDMI_FILTER_TYPE_KV_SUBSTRING.
+	 */
+	struct m0_conf_pver          *ff_dix_pver;
+	/**
+	 * Substrings to match for M0_FDMI_FILTER_TYPE_KV_SUBSTRING filter.
+	 * It might be no string to match, in this case this field should be a
+	 * pointer to a single NULL value and every FDMI record will match it.
+	 */
+	const char                   **ff_substrings;
```

Comment:
Please add comments like:
```
* NULL-terminated array of C strings.
```
So, we know how to determine the size of this array.
Or we need a size/number of the array items.

Reply:
Done.